### DISCUSSION TITLE
why `fulfillRandomWords` function is internal?

### DISCUSSION THREAD

**Discussion by fabriziogianni7** [Cyfrin/foundry-full-course-cu#1541] [General]
# Doubt
Why [`fulfillRandomWords` function](https://github.com/Cyfrin/foundry-smart-contract-lottery-f23/blob/81c9cf13d12d719964aecae7174c6e359492a774/src/Raffle.sol#L169C7-L169C15) is internal?

It's supposed to be called by the coordinator right? The coordinator is another contract , so this function should be external IMO.

Thoughts?

**Comment by fabriziogianni7** [comment]
Ok, I see that it's inheriting fromVRFConsumerBaseV2 so it makes sense

**Comment by aminsammara** [comment]
I had the same question. Are you able to elaborate why it makes sense since it's inheriting from `VRFConsumerBaseV2` ?

**Reply by aminsammara** [reply]
I think I found an answer here https://studygroup.moralis.io/t/inheritance-reading-assignment/9649/902
